src/util/dev/server.ts

```diff
@@ -306,20 +314,30 @@ export default class DevServer {
       return this.cachedNowJson;
     }
 
-    this.output.debug('Reading `now.json` file');
-    const nowJsonPath = getNowJsonPath(this.cwd);
+    const pkg = await this.getPackageJson();
 
     // The default empty `now.json` is used to serve all files as static
     // when no `now.json` is present
     let config: NowConfig = this.cachedNowJson || { version: 2 };
 
+    // We need to delete these properties for zero config to work
+    // with file changes
+    if (this.cachedNowJson) {
+      delete this.cachedNowJson.builds;
+      delete this.cachedNowJson.routes;
+    }
+
     try {
+      this.output.debug('Reading `now.json` file');
+      const nowJsonPath = getNowJsonPath(this.cwd);
       config = JSON.parse(await fs.readFile(nowJsonPath, 'utf8'));
     } catch (err) {
       if (err.code === 'ENOENT') {
-        this.output.note(
-          'No `now.json` file present, serving all files as static'
-        );
+        if (pkg === null) {
+          this.output.note(
+            'No `now.json` file present, serving all files as static'
```

Comment:
Should we adjust this message to say `No now.json or package.json file present`?

Reply:
@TooTallNate I will already output that there is no package.json in this line: https://github.com/zeit/now-cli/pull/2498/files#diff-81b0317d818fc86303285d3ed31c8dd6R400